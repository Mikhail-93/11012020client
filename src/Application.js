import {AbstractSync, Kopnik} from "./models";

export default class Application {
    static getInstance() {
        if (!this.instance) {
            this.instance = new this()
        }
        return this.instance
    }

    user = null

    constructor() {
        /**
         * Кэш моделей
         * @type {Array}
         */
        // пока не понятно зачем приложению кэш.
        // он вероятно будет сильно торможить приложение из-за реактивности App.$data в которой он находится
        // хотя он же не в App.$data а в App.$options !!
        // уже в дата потому тчо в options не реактивен
        //this.models = AbstractSync.cache

        /**
         * Идентификатор раздела
         * Равен названиею соответствующего компонента
         *
         * @type {string} Map | Profile | Thanks
         */
        this.SECTION = "Map"

        this.user = undefined

        this.initUser()
    }

    getSharedState() {
        return {
            SECTION: this.SECTION
        }
    }

    setState(state) {
        if (state.SECTION) {
            this.SECTION = state.SECTION
        }
    }

    /**
     * Инициализирует ползователя при запуске приложения
     *
     * @returns {Promise<void>}
     */
    async initUser() {
        let vkUser = localStorage.getItem("vkUser")
        if (vkUser) {
            vkUser = JSON.parse(vkUser)
            let user //= await Kopnik.getVkUserStatus(vkUser.uid)
            if (vkUserStatus) {
                this.user= Kopnik.merge(vkUserStatus.user)
            } else {
                this.setVkUser(vkUser)
                this.SECTION="Profile"
            }
        }
    }

    /**
     * Учстанавливает пользователем объект из VK.init::success()
     * @param vkUser
     * @returns
     */
    setVkUser(vkUser){
        this.user= new Kopnik
        this.user.merge({
            hash: vkUser.hash,
            uid: vkUser.uid,
            firstname: vkUser.first_name,
            surname: vkUser.last_name,
            photo: vkUser.photo,
            smallPhoto: vkUser.photo_rec,
        })
        localStorage.setItem("vkUser", JSON.stringify(vkUser))
    }
}
