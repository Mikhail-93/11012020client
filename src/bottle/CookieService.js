/**
 * Предоставляет куки
 * По умолчанию куки из конфига config.di.coockie
 * Умеет пуш-попить куки, отходя от конфига при этом, полезно во время сквозных тестов
 * @example coockieService.push(1)
 * @example coockieService.pop()
 */
export default class CookieService {
    constructor(config) {
        this.stack=[null]
    }
    push(){
        return this.stack.push(this.cookie)
    }
    pop(){
        return this.stack.pop()
    }
    get cookie (){
        if (this.stack.length) {
            return this.stack[this.stack.length - 1]
        }
        else{
            return null
        }
    }
    set cookie (value){
        this.stack[this.stack.length-1]=value
    }
}
