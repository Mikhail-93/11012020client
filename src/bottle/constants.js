/**
 * Created by alexey2baranov on 8/20/16.
 *
 * домены и редикт урлы
 * https://vk.com/apps?act=manage
 * дебаг на телефоне
 * https://developers.google.com/web/tools/chrome-devtools/remote-debugging?utm_source=dcc&utm_medium=redirect&utm_campaign=2016q3
 */
let constants = {
  "development": {
    sw: {
      delay: 15000,
    },
    messenger: {
      // идентификатор нашего приложения ВК
      // clientId: 7210289,
      // страница, на которую нужно перейти для аутентификации
      loginUrl: 'http://localhost:8081/connect/vkontakte',
      // с мобилки
      // loginUrl: 'http://192.168.43.84:8081/connect/vkontakte'
    },
    api: {
      path: "http://localhost:8081/api"
      // с мобилки
      // path: "http://192.168.43.84:8081/api"
    },
    di: {
      fetch: true,
      cookie: false
    }
  },
  "test": {
    sw: {
      delay: Number.MAX_SAFE_INTEGER,
    },
    messenger: {
      // clientId: 7210289,
      loginUrl: 'http://localhost:8082/connect/vkontakte',
      // redirectUrl: 'http://localhost:8082/login/check-vk'
    },
    api: {
      path: "http://localhost:8082/api"
    },
    di: {
      fetch: false,
      cookie: true
    }
  },
  "staging": {
    sw: {
      delay: 20000,
    },
    messenger: {
      loginUrl: 'https://staging.kopnik.org/connect/vkontakte',
      // clientId: 7210289,
      // redirectUrl: 'https://staging.kopnik.org/login/check-vk'
    },
    api: {
      path: "https://staging.kopnik.org/api"
    },
    di: {
      fetch: true,
      cookie: false
    }
  }
}

export default constants
