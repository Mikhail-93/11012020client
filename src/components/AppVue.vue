<template>
    <v-app id="inspire">
        <v-navigation-drawer v-model="drawer" app>
            <router-link to="/profile" class="cursor-pointer" tag="div">
                <kopnik v-if="$root.$data.app.user" v-model="$root.$data.app.user" to="/profile"
                        :avatar-size="150" class="flex-grow-1">
                </kopnik>
            </router-link>
            <v-list :dense="false" subheader>
                <v-list-item link to="/">
                    <v-list-item-action>
                        <v-icon>mdi-home-city</v-icon>
                    </v-list-item-action>
                    <v-list-item-content>
                        <v-list-item-title>Карта</v-list-item-title>
                    </v-list-item-content>
                </v-list-item>
                <v-list-item link>
                    <v-list-item-action>
                        <v-icon>mdi-account-multiple</v-icon>
                    </v-list-item-action>
                    <v-list-item-content>
                        <v-list-item-title>Моя десятка</v-list-item-title>
                    </v-list-item-content>
                </v-list-item>
                <v-list-item link>
                    <v-list-item-action>
                        <v-icon>mdi-chat</v-icon>
                    </v-list-item-action>
                    <v-list-item-content>
                        <v-list-item-title>Чат десятки</v-list-item-title>
                    </v-list-item-content>
                </v-list-item>
                <v-list-item link>
                    <v-list-item-action>
                        <v-icon>mdi-chat</v-icon>
                    </v-list-item-action>
                    <v-list-item-content>
                        <v-list-item-title>Чат старшины</v-list-item-title>
                    </v-list-item-content>
                </v-list-item>
                <v-list-item link to="/witness">
                    <v-list-item-action>
                        <v-icon>mdi-human-greeting</v-icon>
                    </v-list-item-action>
                    <v-list-item-content>
                        <v-list-item-title>Заявки на вступление</v-list-item-title>
                    </v-list-item-content>
                    <v-list-item-action>
                        <v-icon>mdi-account-question</v-icon>
                    </v-list-item-action>
                </v-list-item>
                <v-list-item link to="/thanks">
                    <v-list-item-action>
                        <v-icon>mdi-heart</v-icon>
                    </v-list-item-action>
                    <v-list-item-content>
                        <v-list-item-title>Благодарность</v-list-item-title>
                    </v-list-item-content>
                </v-list-item>
                <v-list-item link>
                    <v-list-item-action>
                        <v-icon>mdi-help-circle</v-icon>
                    </v-list-item-action>
                    <v-list-item-content>
                        <v-list-item-title>Подсказка</v-list-item-title>
                    </v-list-item-content>
                </v-list-item>
                <v-divider></v-divider>
                <v-list-item link>
                    <v-list-item-action>
                        <v-icon>mdi-location-exit</v-icon>
                    </v-list-item-action>
                    <v-list-item-content>
                        <v-list-item-title>Выход</v-list-item-title>
                    </v-list-item-content>
                </v-list-item>
            </v-list>
        </v-navigation-drawer>

        <v-app-bar
                app
                color="indigo"
        >
            <v-app-bar-nav-icon @click.stop="drawer = !drawer"/>
            <v-toolbar-title>kopnik.org</v-toolbar-title>
        </v-app-bar>

        <v-content>
            <v-container class="fill-height" fluid>
                <!--                <Auth v-if="!app.user" @login="login_login" class="d-flex justify-center align-center"
                                      style="position:absolute; left:0; top20:0; z-index: 1000; right: 0; bottom: 0;"></Auth>-->
                <!--                <div :is="app.SECTION" style="flex-grow: 1"></div>-->
                <keep-alive>
                    <router-view></router-view>
                </keep-alive>
            </v-container>
        </v-content>
    </v-app>
</template>

<script>

    import Application from "../Application";
    import Kopnik from "./KopnikVue";

    export default {
        components: {
            Kopnik,
        },
        props: {
            source: String,
        },
        data: (() => {
            let app = global.app
            // app.authenticate()

            return {
                app,
                drawer: null,
                center: [47.413220, -1.219482],
                zoom: 14,
                tileProviders: [
                    {
                        name: "OpenStreetMap",
                        visible: false,
                        url: "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
                        attribution: 'Map data: &copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>, <a href="http://viewfinderpanoramas.org">SRTM</a> | Map style: &copy; <a href="https://opentopomap.org">OpenTopoMap</a> (<a href="https://creativecommons.org/licenses/by-sa/3.0/">CC-BY-SA</a>)',
                        token: null
                    },
                    {
                        name: "Dark",
                        visible: true,
                        url: "https://{s}.basemaps.cartocdn.com/dark_all/{z}/{x}/{y}{r}.png",
                        attribution: '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>',
                        subdomains: "abcd",
                        token: null
                    },
                    {
                        name: "GIScience",
                        visible: false,
                        url: "https://maps.heigit.org/openmapsurfer/tiles/roads/webmercator/{z}/{x}/{y}.png",
                        attribution: 'Imagery from <a href="http://giscience.uni-hd.de/">GIScience Research Group @ University of Heidelberg</a> | Map data &copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors',
                        subdomains: "abcd",
                        token: null
                    }
                ],
            }
        }),
        computed: {
            user() {
                return this.$options.app.user
            }
        },
        methods: {
            onLocationClick() {
            },
            onKopnikClick(kopnik) {
                alert("Open face details ")
            },
            login_login() {

            }
        },
        mounted() {
            this.$nextTick(() => {
                // this.map=
                // this.$refs.map.mapObject.setView([51.505, -0.09], 13)

            })
        }
    }
</script>

<style scoped>
    .cursor-pointer{
        cursor: pointer;
    }
</style>
