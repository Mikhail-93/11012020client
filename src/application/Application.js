import get from "keypather/get"
import has from "keypather/has"

import AsyncLock from 'async-lock'
import {AbstractSync, Kopnik, Kopa} from "../models";
import {KopnikApiError, KopnikError} from "../KopnikError";
import once from "../decorators/once";
import SquadAnalyzer from "../SquadAnalyzer";
import fetchIntercept from 'fetch-intercept'
import {container} from "../bottle/bottle";
import {LatLng, LatLngBounds} from 'leaflet'
import Main from "./Main";
import messages from "../locales";

export default class Application {
    constructor(logger) {
        this.logger = logger.getLogger('Application')
        /**
         * Кэш моделей
         * @type {Array}
         */
        // пока не понятно зачем приложению кэш.
        // он вероятно будет сильно торможить приложение из-за реактивности App.$data в которой он находится
        // хотя он же не в App.$data а в App.$options !!
        // уже в дата потому тчо в options не реактивен
        //this.models = AbstractSync.cache

        /**
         * Идентификатор раздела
         * Равен названиею соответствующего компонента
         *
         * @type {string} Map | Profile | Thanks
         */
        this.section = Application.Section.Main
        this.sectionLocker = new AsyncLock

        /**
         * 0 потому что на undefined не срабатывает watcher,
         * а null занят под "пользователь не аутентифицирован"
         * @type {Kopnik}
         */
        this.user = undefined

        /**
         * Сообщения пользователю
         * @type {String[]}
         */
        this.infos = []
        /**
         * Обишки приложения
         * @type {Error[]}
         */
        this.errors = []
        this.sections = {
            main: new Main(this)
        }
    }

    async onerror(err) {
        if (err.code === 401) {
            await container.application.lockSection(async () => {
                await application.setSection(Application.Section.Main)
            })
            this.user= null
            console.info('prevent 401 error', err)
        } else {
            this.logger.error(err)
            if (err instanceof KopnikApiError){
                this.logger.info(err.url)
            }
            this.errors.push(err)
            // throw err
        }
    }

    getMessage(message) {
        for(const eachLocaleName of [container.localeManager.currentLocale.name, 'ru']) {
            if (has(container.messages[eachLocaleName], message)) {
                const result = get(container.messages[eachLocaleName], message)
                return result
            }
        }
        throw new KopnikError('No message provided: ' + message, 2)
    }

    /**
     * Эксклюзивный доступ к section
     * @param {function} callback
     * @returns {Promise<any>}
     */
    lockSection(callback) {
        return this.sectionLocker.acquire('section', callback)
    }

    /**
     * Устанавливить секцию
     * Если секция не может быть установлена, возвращает иную
     *
     * @param {String} section
     * @returns {Promise<string>} Установленная секция
     */
    async setSection(section) {
        let result
        if (this.section === section) {
            return section
        }
        switch (section) {
            case Application.Section.Profile:
            case Application.Section.Witness:
            case Application.Section.Ten:
                if (await this.resolveUser()) {
                    if (section === Application.Section.Profile || this.user.status === Kopnik.Status.CONFIRMED) {
                        result = section
                    } else {
                        result = await this.setSection(Application.Section.Main)
                    }
                } else {
                    result = await this.setSection(Application.Section.Main)
                }
                break
            case Application.Section.Main:
            case Application.Section.Thanks:
                result = section
                break
            default:
                throw new KopnikError('Wrong route', 666)
        }
        // this.logger.info('move', this.section, '->', section)
        return this.section = result
    }


    getSharedState() {
        return {
            SECTION: this.section
        }
    }

    setState(state) {
        if (state.section) {
            this.section = state.section
        }
    }

    /**
     * Определиться с пользователем. Или он некий копник, или он null, то есть незарегистрирован на сервере
     * @returns {Promise<Kopnik>}
     */
    async resolveUser() {
        if (this.user === undefined) {
            await this.authenticate()
        }
        return this.user
    }

    /**
     * Инициализирует ползователя при запуске приложения
     *
     * @returns {Promise<void>}
     */
    @once
    async authenticate() {
        try {
            let userAsPlain = (await Kopnik.api('get?ids='))[0]
            this.user = await Kopnik.get(userAsPlain.id)
            container.localeManager.currentLocale= this.user.locale
            this.logger.info('user authenticated', this.user)
        } catch (err) {
            if ((err instanceof KopnikApiError) && err.message.match(/no.+aut/i)) {
                 this.user = null
                this.logger.info('user not authenticated')
            } else {
                throw err
            }
        }
    }

    /**
     * Инициализирует ползователя при запуске приложения
     *
     * @returns {Promise<void>}
     */
    @once
    async logout() {
        await this.lockSection(() => {
            this.setSection(Application.Section.Main)
        })
        this.sections.main.selected= null
        await container.api('logout')
        this.user = null
    }
}

/**
 * @enum {string}
 */
Application.Section = {
    Main: 'Main',
    Profile: 'Profile',
    Witness: 'Witness',
    Thanks: 'Thanks',
    Ten: 'Ten',
    Help: 'Help',
}
