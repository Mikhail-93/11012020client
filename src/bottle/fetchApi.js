import {container} from './bottle'
import _ from 'lodash'
import jsonToFormData from 'json-form-data'
import {KopnikApiError} from "../KopnikError";


export default async function fetchApi(url, options = {}) {
    const defaultOptions = {
            credentials: 'include',
            headers: {
                Accept: 'application/json',
                'Content-Type': 'text/plain',
            }
        },
        cookieOptions = (container.config.di.cookie && container.cookieService.cookie) ? {headers: {Cookie: container.cookieService.cookie}} : null

    options = _.merge({}, defaultOptions, cookieOptions, options)
    if (options.body) {
        options.body = JSON.stringify(options.body)
    }
    // container.logger.warn(options)
    // console.log(container.config.api.path)
    let fullUrl = `${container.config.api.path}/${url}`.replace(/\w+\/\.\.\//, '')

    try {
        // console.log(url, options)
        var response = await fetch(fullUrl, options)
        // Пропал 4G
    } catch (err) {
        throw new KopnikApiError(err.message, null, fullUrl)
    }
    // Не найдена страница или синтаксическая ошибка в веб-сервисе
    if (!response.ok) {
        throw new KopnikApiError(response.statusText, response.status, fullUrl)
    }

    let cookie = response.headers.get('set-cookie')
    if (container.config.di.cookie && cookie) {
        cookie = cookie.match(/(\w+=(\w|\d)+)/)[0]
        container.cookieService.cookie = cookie
    }
    // container.logger.warn(cookie, response.headers)
    let result
    switch (response.headers.get('Content-Type')) {
        case 'application/json':
            // result = await response.text()
            result = await response.json()
            // Не авторизован/Нет такого пользователя
            if (result.error) {
                console.log(result.error)
                throw new KopnikApiError(result.error.error_msg, result.error.error_code, fullUrl)
            }
            break
        default:
            result = await response.text()
    }

    return result.response
}
