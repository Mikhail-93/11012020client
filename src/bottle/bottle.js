import Bottle from "bottlejs"
import _ from "lodash"
// import 'json-form-data'
import * as loglevel from "loglevel";

import Application from "../application/Application"
import constants from './constants'
import api from "../api";
import CookieService from "./CookieService"
import MK from "../mk/mk";
import LocaleManager from "../locales/LocaleManager";
import ru from '../locales/ru'
import en from '../locales/en'

Bottle.config.strict = true
const bottle = new Bottle()

bottle.factory('logger', function loggerFactory() {
    //все плагины ломают стектрейс консоли. то есть невозможно увидеть из какого файла и какой строки был вызван лог!
    // LoglevelPluginPrefix.reg(loglevel)
    // LoglevelPluginPrefix.apply(loglevel, {template: "%t [%l] %n: "})
    loglevel.setLevel(loglevel.levels.DEBUG)
    if (process.env.NODE_ENV == "test") {
        loglevel.setLevel(loglevel.levels.DEBUG)
        // LoglevelPluginToString.apply(loglevel, {})
    }
// Be sure to call setLevel method in order to apply plugin
// loglevel.getLogger("StateManager").setLevel("info")
    return loglevel
})

bottle.constant('messages', {
    ru,
    en,
})

bottle.factory('env', function env() {
    return process.env.NODE_ENV || 'development'
})
bottle.service('localeManager', function localeManager() {
    const localeManager = new LocaleManager()
    localeManager.currentLocale = 'ru'//container.env === 'development' ? 'en' : 'ru'
    return localeManager
})
bottle.service('cookieService', CookieService, 'constants')
bottle.factory('api', function apiFactory(container) {
    if (!container.constants.di.fetch) {
        throw new Error('not supported')
    }
    return api
})
bottle.factory('VK', function vkFactory(container) {
    return process.env.NODE_ENV === 'test' ? MK : global.VK
})
bottle.factory('constants', function constantsFactory() {
    if (!process.env.NODE_ENV) {
        throw new Error("NODE_ENV is not defined")
    }

    let local = {}//require("./local.js")
    let result = _.merge({}, constants, local)[process.env.NODE_ENV]

    return result
})
if (!Application) {
    throw new Error("Application is undefined in bottle")
}
bottle.service('application', Application, 'logger')

/**
 * @callback fetch
 * @param {string} url
 * @param {Object?} options
 */
/**
 * @type {Object}
 *
 * @property {loglevel} logger
 * @property {Application} application
 * @property {Location} Location
 * @property {fetch} api
 * @property {MK.class} VK
 * @property {CookieService} cookieService
 * @property {Object} constants
 * @property {Object} constants.api
 * @property {String} constants.api.path
 * @property {Object} constants.di
 * @property {Boolean} constants.di.fetch
 * @property {Number} constants.di.cookie
 * @property {Number} constants.messenger.clientId
 * @property {String} constants.messenger.loginUrl
 * @property {String} constants.messenger.redirectUrl
 * @property {String} constants.sw
 * @property {String} constants.sw.delay
 * @property {LocaleManager} localeManager
 * @property {string} env
 * @property {{ru, en}} messages
 */
const container = bottle.container
export {bottle, container}
