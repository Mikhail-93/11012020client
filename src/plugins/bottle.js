import Bottle from "bottlejs"
import _ from "lodash"
import 'json-form-data'
import * as loglevel from "loglevel";

import Application from "../Application"
import config from '../../config'
import fetchApi from "../bottle/fetchApi";
import fetchApiMock from "../bottle/fetchApi.mock";

Bottle.config.strict = true
const bottle = new Bottle()

/**
 * Настраивает контейнер
 * Для тестов необходим сброс и перенастройка чтобы соблюсти изолированность тестов
 *
 * @param {Object} options параметры по умолчанию соответствуют наиболее приемлемым для окружения значениям окружению
 * @param {Number=} options.cookie
 * @param {Boolean=} options.fetch
 */
Bottle.prototype.setup = function (options) {
    bottle.factory('defaultFetchApiOptions', function defaultFetchApiOptionsFactory(container) {
        const cookies = {
            1: 'PHPSESSID=user-1',
            2: 'PHPSESSID=eqlae6a1grpvcirpemfvebf806',
            3: 'PHPSESSID=user-3',
            4: 'PHPSESSID=user-4',
            5: 'PHPSESSID=user-5',
            6: 'PHPSESSID=user-6',
            7: 'PHPSESSID=user-7',
            8: 'PHPSESSID=user-8',
            9: 'PHPSESSID=user-9',
            10: 'PHPSESSID=user-10',
        }
        const result = {}
        if (container.config.di.cookie) {
            result.headers = {
                cookie: cookies[container.config.di.cookie]
            }
        }
        // console.log(result)
        return result
    })
    bottle.factory('fetchApi', function fetchApiFactory(container) {
        return container.config.di.fetch ? fetchApi : fetchApiMock
    })
    bottle.factory('config', function configFactory() {
        if (!process.env.NODE_ENV) {
            throw new Error("NODE_ENV is not defined");
        }

        let local = {}//require("./local.js")
        let result = _.merge({}, config, local)[process.env.NODE_ENV]

        return result
    })
    if (!Application){
        throw new Error("Application is undefined in bottle")
    }
    bottle.service('application', Application, 'logger')
    bottle.factory('logger', function loggerFactory() {
        //все плагины ломают стектрейс консоли. то есть невозможно увидеть из какого файла и какой строки был вызван лог!
        // LoglevelPluginPrefix.reg(loglevel)
        // LoglevelPluginPrefix.apply(loglevel, {template: "%t [%l] %n: "})
        loglevel.setLevel(loglevel.levels.DEBUG)
        if (process.env.NODE_ENV == "test") {
            loglevel.setLevel(loglevel.levels.WARN)
            // LoglevelPluginToString.apply(loglevel, {})
        }
// Be sure to call setLevel method in order to apply plugin
// loglevel.getLogger("StateManager").setLevel("info")
        return loglevel
    })
    Object.assign(this.container.config.di, options)
}

bottle.setup()

/**
 * @callback fetch
 * @param {string} url
 * @param {Object} options
 */
/**
 * @type {Object}
 *
 * @property {loglevel} logger
 * @property {Application} application
 * @property {Location} Location
 * @property {fetch} fetchApi
 * @property {Object} defaultFetchApiOptions
 * @property {Object} defaultFetchApiOptions.headers
 * @property {Object} defaultFetchApiOptions.headers.cookie
 * @property {Object} config
 */
const container = bottle.container
export {bottle, container}
