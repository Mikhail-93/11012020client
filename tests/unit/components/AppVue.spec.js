import Vue from 'vue'
import {i18n, vuetify, routerFactory} from "../../test-setup";
import AppVue from "../../../src/components/AppVue";
import {bottle, container} from "../../../src/plugins/bottle";
import flushPromises from "flush-promises";
import Application from "../../../src/Application";

describe('unit conponents AppVue', () => {
    let vm,
        application

    beforeEach(() => {
        bottle.resetProviders(['application', 'cookieService'])
        application = container.application
        vm = new Vue({
            ...AppVue,
            i18n,
            vuetify,
            router: routerFactory()
        })
    })

    describe('anonymous', () => {
        it('render', async () => {
            application.authenticate()
            vm.$mount()
            await flushPromises()
        })

        describe('enter', () => {
            it('/', async () => {
                application.authenticate()
                vm.$mount()
                await flushPromises()
                expect(vm.$el.textContent).toContain('Войти через ВКонтакте')
                expect(application.section).toBe(Application.Section.Main)
            })
            it('/profile', async () => {
                application.authenticate()
                vm.$mount()
                try {
                    await vm.$router.push({name: Application.Section.Profile})
                    throw new Error('should not be hire')
                } catch (err) {
                    expect(err).toBe(false)
                    await flushPromises()
                    expect(vm.$router.currentRoute.name).toBe(Application.Section.Main)
                    expect(application.section).toBe(Application.Section.Main)
                    expect(vm.$el.textContent).toContain('Войти через ВКонтакте')
                }
            })
            it('/Join', async () => {
                application.authenticate()
                vm.$mount()
                try {
                    await vm.$router.push({name: Application.Section.Join})
                    throw new Error('should not be hire')
                } catch (err) {
                    expect(err).toBe(false)
                    await flushPromises()
                    expect(vm.$router.currentRoute.name).toBe(Application.Section.Main)
                    expect(application.section).toBe(Application.Section.Main)
                    expect(vm.$el.textContent).toContain('Войти через ВКонтакте')
                }
            })
        })
    })

    // ожидаем users/list чтобы пожно было под разными пользователями заходить
    describe.skip('status=new', () => {
        beforeEach(async () => {
            await login(2)
        })

        it('render', async () => {
            application.authenticate()
            vm.$mount()
            await flushPromises()
        })

        describe('enter', () => {
            it('/', async () => {
                application.authenticate()
                vm.$mount()
                await flushPromises()
                expect(vm.$el.textContent).not.toContain('Войти через ВКонтакте')
                expect(application.section).toBe(Application.Section.Main)
            })
            it('/profile', async () => {
                application.authenticate()
                vm.$mount()
                    await vm.$router.push({name: Application.Section.Profile})
                    await flushPromises()
                    expect(vm.$router.currentRoute.name).toBe(Application.Section.Profile)
                    expect(application.section).toBe(Application.Section.Profile)
                    expect(vm.$el.textContent).toContain('Язык')
            })
            it('/Join', async () => {
                application.authenticate()
                vm.$mount()
                try {
                    await vm.$router.push({name: Application.Section.Join})
                    throw new Error('should not be hire')
                } catch (err) {
                    expect(err).toBe(false)
                    await flushPromises()
                    expect(vm.$router.currentRoute.name).toBe(Application.Section.Main)
                    expect(application.section).toBe(Application.Section.Main)
                    expect(vm.$el.textContent).toContain('Войти через ВКонтакте')
                }
            })
        })
    })

})
