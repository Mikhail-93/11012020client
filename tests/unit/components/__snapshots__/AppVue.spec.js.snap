// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit conponents AppVue anonymous renders 1`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="inspire"
>
  <div
    class="v-application--wrap"
  >
    <!---->
     
    <!---->
     
    <!---->
     
    <!---->
     
    <main
      class="v-content"
      style="padding: 0px 0px 0px 0px;"
    >
      <div
        class="v-content__wrap"
      >
        <div
          class="row fill-height pa-0 flex-column flex-nowrap align-center"
          login=""
          style="position: fixed; top: 0px; left: 0px; right: 0px; z-index: 100;"
        >
          <div
            class="col"
          />
           
          <div
            class="v-avatar"
            style="height: 170px; min-width: 170px; width: 170px;"
          >
            <div
              class="v-responsive v-image"
            >
              <div
                class="v-image__image v-image__image--preload v-image__image--cover"
                style="background-position: center center;"
              />
              <div
                class="v-responsive__content"
              />
            </div>
          </div>
           
          <div
            class="pa-1 tagline text-center"
          >
            <span>
              КОПНОЕ ПРАВО
            </span>
             
            <div
              style="font-size: smaller;"
            >
              в телефоне каждого славянина
            </div>
          </div>
           
          <div
            class="d-flex align-end col"
            style="width: inherit;"
          >
            <button
              class="mb-12 v-btn v-btn--contained theme--light v-size--default primary"
              type="button"
            >
              <span
                class="v-btn__content"
              >
                Войти через ВКонтакте
              </span>
            </button>
          </div>
        </div>
         
        <div
          class="container fill-height pa-0 k-content container--fluid k-main"
        >
          <div
            class="v-dialog__content"
            role="document"
            style="z-index: 0;"
          >
            <div
              class="v-dialog v-dialog--persistent v-bottom-sheet v-bottom-sheet--inset"
              style="max-width: auto; display: none;"
            />
          </div>
          <div
            class="vue2leaflet-map leaflet-container leaflet-grab leaflet-touch-drag"
            style="z-index: 0; position: relative;"
            tabindex="0"
          >
            <div />
            <div />
            <div />
             
            <!---->
            <div
              class="leaflet-pane leaflet-map-pane"
              style="left: 0px; top: 0px;"
            >
              <div
                class="leaflet-pane leaflet-tile-pane"
              >
                <div
                  class="leaflet-layer "
                  style="z-index: 1;"
                >
                  <div
                    class="leaflet-tile-container leaflet-zoom-animated"
                    style="z-index: 18; left: 0px; top: 0px;"
                  >
                    <img
                      alt=""
                      class="leaflet-tile"
                      role="presentation"
                      src="https://a.tile.openstreetmap.org/2/2/1.png"
                      style="width: 256px; height: 256px; left: -107px; top: -64px;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="leaflet-pane leaflet-shadow-pane"
              />
              <div
                class="leaflet-pane leaflet-overlay-pane"
              />
              <div
                class="leaflet-pane leaflet-marker-pane"
              />
              <div
                class="leaflet-pane leaflet-tooltip-pane"
              />
              <div
                class="leaflet-pane leaflet-popup-pane"
              />
            </div>
            <div
              class="leaflet-control-container"
            >
              <div
                class="leaflet-top leaflet-left"
              >
                <div
                  class="leaflet-control-zoom leaflet-bar leaflet-control"
                >
                  <a
                    aria-label="Zoom in"
                    class="leaflet-control-zoom-in"
                    href="#"
                    role="button"
                    title="Zoom in"
                  >
                    +
                  </a>
                  <a
                    aria-label="Zoom out"
                    class="leaflet-control-zoom-out"
                    href="#"
                    role="button"
                    title="Zoom out"
                  >
                    −
                  </a>
                </div>
              </div>
              <div
                class="leaflet-top leaflet-right"
              >
                <div
                  class="leaflet-control"
                />
              </div>
              <div
                class="leaflet-bottom leaflet-left"
              />
              <div
                class="leaflet-bottom leaflet-right"
              >
                <div
                  class="leaflet-control-attribution leaflet-control"
                >
                  <a
                    href="https://leafletjs.com"
                    title="A JS library for interactive maps"
                  >
                    Leaflet
                  </a>
                   | Map data: © 
                  <a
                    href="http://www.openstreetmap.org/copyright"
                  >
                    OpenStreetMap
                  </a>
                  , 
                  <a
                    href="http://viewfinderpanoramas.org"
                  >
                    SRTM
                  </a>
                   | Map style: © 
                  <a
                    href="https://opentopomap.org"
                  >
                    OpenTopoMap
                  </a>
                   (
                  <a
                    href="https://creativecommons.org/licenses/by-sa/3.0/"
                  >
                    CC-BY-SA
                  </a>
                  )
                </div>
              </div>
            </div>
             
            <!---->
             
            <!---->
             
            <!---->
             
              
          </div>
           
          <!---->
           
          <div
            class="v-dialog__container"
            role="dialog"
          />
        </div>
      </div>
    </main>
  </div>
</div>
`;
